webapp/test/LoginForm.spec.js

```diff
@@ -0,0 +1,275 @@
+import '@testing-library/jest-dom'
+import 'regenerator-runtime'
+
+import { createLocalVue, shallowMount } from '@vue/test-utils'
+
+import VueApollo from 'vue-apollo'
+import { createMockClient } from 'mock-apollo-client'
+import login from '@/apollo/mutations/login'
+
+import Vuex from 'vuex'
+
+import LoginForm from '@/components/LoginForm'
+import { actions, getters, mutations, state } from '@/store/auth'
+import { InMemoryCache } from '@apollo/client'
+
+import jwt_decode from 'jwt-decode'
+
+jest.mock('jwt-decode')
+
+describe('LoginForm', () => {
+  const localVue = createLocalVue()
+  localVue.use(VueApollo)
+  localVue.use(Vuex)
+
+  let wrapper, mockClient, apolloProvider, store
+
+  const createLoginForm = async (handlers, appOptions) => {
+    mockClient = createMockClient({ cache: new InMemoryCache() })
+    apolloProvider = new VueApollo({
+      defaultClient: mockClient
+    })
+
+    const requestHandlers = {
+      loginHandler: null,
+      ...handlers
+    }
+    mockClient.setRequestHandler(
+      login,
+      requestHandlers.loginHandler
+    )
+
+    store = new Vuex.Store({
+      modules: {
+        auth: {
+          namespaced: true,
+          state,
+          getters,
+          actions,
+          mutations
+        }
+      }
+    })
+
+    wrapper = shallowMount(LoginForm, {
+      localVue,
+      apolloProvider,
+      store,
+      stubs: {
+        NuxtLink: true,
+      },
+      ...appOptions
+    })
+    await wrapper.vm.$nextTick()
+  }
+
+  afterEach(() => {
+    wrapper.destroy()
+    mockClient = null
+    apolloProvider = null
+    store = null
+  })
+
+  describe('when no user is logged in', () => {
+    it('should show the login form', async () => {
+      await createLoginForm()
+      await wrapper.vm.$nextTick()
+
+      expect(wrapper.find('#form-login').element).not.toBeEmptyDOMElement()
+      expect(wrapper.html()).toMatchSnapshot()
+    })
+
+    it('should only enable the submit button if both email and password are provided', async () => {
+      await createLoginForm()
+      await wrapper.vm.$nextTick()
+
+      const loginButton = wrapper.find('input[value="Login"]').element
+
+      expect(loginButton).toBeDisabled()
+
+      wrapper.find('#input-email').setValue('someEmail@addr.es')
+      await wrapper.vm.$nextTick()
+
+      expect(loginButton).toBeDisabled()
+
+      wrapper.find('#input-password').setValue('someSecret')
+      await wrapper.vm.$nextTick()
+
+      expect(loginButton).toBeEnabled()
+
+      wrapper.find('#input-password').setValue('')
+      await wrapper.vm.$nextTick()
+
+      expect(loginButton).toBeDisabled()
```

Comment:
Try to have one expectation per test-case for readability and to avoid false-negatives (because of state altered in some special order). You can split this into multiple test cases and use `beforeEach` for a shared setup.